### PR TITLE
Supported workspace trust

### PR DESCRIPTION
**Pull request by tianyiw2013**
I try to merge the modified parts of my branch into your master.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,7 +122,7 @@ and can be triggered by typing @ then another characted (Provided your vscode se
 | @abstract                    | @abstract                               |
 | @after                       | @after                                  |
 | @afterClass                  | @afterClass                             |
-| @author                      | @author ${1:{{name}}} <${2:{{email}}}>  |
+| @author                      | @author \${1:name} <${2:email}>         |
```
Comment (tianyiw2013) [inline README.md:125]: Markdown escape character

File: package.json [hunk 2]
```diff
@@ -28,6 +28,11 @@
   "bugs": {
     "url": "https://github.com/neild3r/vscode-php-docblocker/issues"
   },
+  "capabilities": {
+    "untrustedWorkspaces": {
+      "supported": true
```
Comment (tianyiw2013) [inline package.json:33]: Supported vscode workspace trust

File: package.json [hunk 3]
```diff
@@ -39,7 +44,7 @@
     ],
     "configuration": {
       "type": "object",
-      "title": "PHP DocBlocker configuration",
+      "title": "PHP DocBlocker",
```
Comment (tianyiw2013) [inline package.json:47]: Shorten title length

File: src/block/function.ts [hunk 4]
```diff
@@ -16,7 +16,7 @@ export default class FunctionBlock extends Block
     /**
      * @inheritdoc
      */
-    protected pattern:RegExp = /^\s*((.*)(protected|private|public))?(.*)?\s*function\s+&?([A-Za-z0-9_]+)\s*\(([^{;]*)/m;
+    protected pattern:RegExp = /^\s*((.*)(protected|private|public))?(.*)?\s*function\s+&?([a-z0-9_]+)\s*\(([^{;]*)/im;
```
Comment (tianyiw2013) [inline src/block/function.ts:19]: ignore case sensitivity. `FUNCTION test()`

File: src/block/class.ts [hunk 5]
```diff
@@ -10,7 +10,7 @@ export default class Class extends Block
     /**
      * @inheritdoc
      */
-    protected pattern:RegExp = /^\s*(abstract|final)?\s*(class|trait|interface)\s+([A-Za-z0-9_]+)\s*/;
+    protected pattern:RegExp = /^\s*(abstract|final)?\s*(class|trait|interface)\s+([a-z0-9_]+)\s*/i;
```
Comment (tianyiw2013) [inline src/block/class.ts:13]: `CLASS TEST{};`

File: src/block.ts [hunk 6]
```diff
@@ -246,7 +246,7 @@ export abstract class Block
         }
 
         // Check for string
-        if (value.match(/^\s*(["'])/) !== null) {
+        if (value.match(/^\s*(["'])/) !== null || value.match(/^\s*<<</) !== null) {
```
Comment (tianyiw2013) [inline src/block.ts:249]:
```php
/** @var string $var  */
$var = <<<EOF
string
EOF;
```

File: src/block.ts [hunk 7]
```diff
@@ -231,7 +231,7 @@ export abstract class Block
         let result:Array<string>;
 
         // Check for bool
-        if (value.match(/^\s*(false|true)\s*$/i) !== null) {
+        if (value.match(/^\s*(false|true)\s*$/i) !== null || value.match(/^\s*\!/i) !== null) {
```
Comment (tianyiw2013) [inline src/block.ts:234]:
```php
/** @var bool $var  */
$var = !0;

/** @var bool $var  */
$var = !'test';
```

**Comment by neild3r** [comment]
If we can just amend, the CHANGELOG for the other fixes then we can merge and release, thank you for your work on this. Sorry I have been away for a while

**Comment by shivapoudel** [comment]
@tianyiw2013 will all the changes and feature requests from your fork will be soon available over this extension?

**Comment by tianyiw2013** [comment]
> @tianyiw2013 will all the changes and feature requests from your fork will be soon available over this extension?

I'm working on it. Please be patient.

**Pull request merged**